### ISSUE TITLE
shovill 1.1.0+galaxy0 failure with compressed FASTQ

### ISSUE THREAD

**Issue by pvanheus** [galaxyproject/tools-iuc#3933]
As can be seen in this history: https://usegalaxy.eu/u/pvanheus/h/shovill-problem the latest version of shovill (version 1.1.0+galaxy0 - as of this writing) throws an error when trying to process compressed FASTQ data.

The specific part of the log showing the error seems to be:

```
[shovill] Running: kmc -sm -m2 -t4 -k21 -ci10 \/data\/dnb05\/galaxy_db\/files\/0\/6\/7\/dataset_06763038\-80b7\-4de5\-8f13\-3eff046f14ff\.dat /data/jwd/main/029/980/29980083/tmp/oHyt1Lfu7Q/kmc /data/jwd/main/029/980/29980083/tmp/oHyt1Lfu7Q 2>&1 | sed 's/^/[kmc] /' | tee -a shovill.log
[shovill] Could not determine genome size from ''
```
This might be an upstream error.

**Comment by pvanheus** [comment]
(should probably also investigate: https://github.com/tseemann/shovill/issues/152)

**Comment by pvanheus** [comment]
The error seems to derive from the fact that `kmc` is somehow getting the `.dat` filenames, i.e. the symbolic link is being resolved before the filename is being passed to kmc.

**Comment by pvanheus** [comment]
followup up some more - the problem comes from https://github.com/tseemann/shovill/blob/master/bin/shovill#L115 - in version 1.0.4 it was using File::Spec->rel2abs https://github.com/tseemann/shovill/blob/v1.0.4/bin/shovill#L124 - the difference in operation of these two functions can be seen using this example:

```
$ ls -l /tmp/foo /tmp/bar
lrwxrwxrwx 1 pvh pvh 8 Sep 10 07:30 /tmp/bar -> /tmp/foo
-rw-rw-r-- 1 pvh pvh 0 Sep 10 07:30 /tmp/foo
$ $ perl -e 'use Cwd; use File::Spec; print Cwd::abs_path("/tmp/bar"),"\n"; print File::Spec->rel2abs("/tmp/bar"),"\n";'
/tmp/foo
/tmp/bar
```

**Comment by pvanheus** [comment]
It has now been several months and there is no response from the upstream tool author. In the interim, support for compressed FASTQ (`fastqsanger.gz,fastqsanger.bz2`) should be removed from this tool.

**Comment by pvanheus** [comment]
Thanks for the suggestion @mvdbeek - addressed in #4378.

**Comment by bernt-matthias** [comment]
Closed via https://github.com/galaxyproject/tools-iuc/pull/4378

Thanks
